.github/workflows/ci_workflows.yml

```diff
@@ -0,0 +1,124 @@
+name: CI
+
+# TODO: cron on a separate YAML?
+on:
+  push:
+  pull_request:
+
+# The following is needed to avoid issues if e.g. Matplotlib tries
+# to open a GUI window.
+env:
+  SETUP_XVFB: True
+
+jobs:
+  code_style:
+    runs-on: ubuntu-latest
+    env:
+      TOXENV: codestyle
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Code style checks
+      run: |
+        echo "TOXENV="$TOXENV
+        tox -v
+
+  initial_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.7', '3.8']
+        # Customize the jobs a little based on Python version
+        include:
+          # Minimal dependencies
+          - python: '3.7'
+            toxenv: py37-test
+            toxargs: -v
+            cc: ''
+          # All optional dependencies
+          - python: '3.8'
+            toxenv: py38-test-alldeps
+            toxargs: --durations=50
+            cc: clang
+    env:
+      TOXENV: ${{ matrix.toxenv }}
+      TOXARGS: ${{ matrix.toxargs }}
+      CC: ${{ matrix.cc }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Fetch tags
+      run: git fetch --prune --unshallow --tags
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    # Write configuration items to standard location to make sure they are
+    # ignored (the tests will fail if not)
+    - name: Before install
+      run: |
+        mkdir -p $HOME/.astropy/config/
+        printf "unicode_output = True\nmax_width = 500" > $HOME/.astropy/config/astropy.cfg
+      shell: bash
+    - name: Install
+      run: |
+        sudo apt-get install language-pack-de tzdata
+        python -m pip install --upgrade tox
+    - name: Run tests
+      run: |
+        echo "CC="$CC
+        echo "TOXENV="$TOXENV
+        echo "TOXARGS="$TOXARGS
+        tox ${TOXARGS}
+
+  # TODO: Maybe this can be part of a matrix too but I just want to show
+  #       stages without porting everything for now.
+  # Full tests with coverage checks.
+  coverage_test:
+    runs-on: ubuntu-latest
+    needs: [code_style, initial_tests]
```

Comment:
I think we could try and keep things simple and not worry about replicating stages for now - we could have a single jobs section with an include matrix?

Reply:
Instead of stages I think we could rely on Actions' fast-fail functionality (default) which means that all jobs stop if one fail. In practice this means if e.g. code style fails it won't waste time on longer builds for example.

Reply:
Up to you. I was just trying to recreate what we had at the time.

Reply:
I think the issue is that the stages makes everything much more verbose. I think I would prefer if we could make things more concise and avoid repetition and then decide if adding stages back are worth it?